### DISCUSSION TITLE
Tiny model going insane

### DISCUSSION THREAD

**Discussion by fillwithjoy1** [openai/whisper#1962] [Q&A] [answered]
Decided to try and transcribe Panda - Desiigner which was a song that hard lyrics so hard to decode that Desiigner himself had to add into Genius (and led to Google allegedly stealing it and Genius fighting Google case). 

except, well its gone insane even when i give it an initial prompt

Python Version: 3.11.7
Package Version: 20231117

![image](https://github.com/openai/whisper/assets/29514446/dfbd82bb-b517-4e9f-b829-2c90c52aeb3c)

![image](https://github.com/openai/whisper/assets/29514446/1e8bbd42-8945-4988-a46d-a111c4e98285)

I suspect this could be due to the song having a guy in the background part of the music making Grraaah and Yeah yeah yeah sounds???

**Comment by Purfview** [comment]
Did you tried it with the latest PR included -> https://github.com/openai/whisper/pull/1838?
And try with this bugfix too -> https://github.com/openai/whisper/pull/1903.

**Reply by fillwithjoy1** [reply]
idek, i just downloaded the latest version from python

**Reply by Purfview** [reply]
Then you don't have any of that.
You need to install Git: https://git-scm.com/downloads   [all default, except in option with PATH select option at the bottom]

To have all latest merged PRs use:
```
python -m pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git
```

To have unmerged #1903 PR [not sure if this is correct]:
```
python -m pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git@refs/pull/1903/head
```

**Reply by Purfview** [reply, marked as answer]
Another alternative is to install from the downloaded ziped repo.

`python -m pip install --upgrade --force-reinstall "whisper-main.zip"` [ downloaded from [https://github.com/openai/whisper](https://github.com/openai/whisper) ]

Or to have https://github.com/openai/whisper/pull/1903 too:
`python -m pip install --upgrade --force-reinstall "whisper-patch-1.zip"` [ downloaded from [https://github.com/Purfview/whisper/tree/patch-1](https://github.com/Purfview/whisper/tree/patch-1) ]

**Reply by fillwithjoy1** [reply]
Seems like that fixed it. Thanks! 
![image](https://github.com/openai/whisper/assets/29514446/f14d393a-9958-44bb-8606-bcfa5f154cd7)

**Reply by Purfview** [reply]
Did you installed with #1903 ["whisper-patch-1.zip"] ?

**Reply by fillwithjoy1** [reply]
Yes I used whisper-patch-1

**Reply by Purfview** [reply]
Dunno if you used exactly this audio -> https://www.youtube.com/watch?v=4NJlUribp3c

I tested aac from it  and on the first try I got almost exactly same hallucination with "come on, come on, come on...".
This very long repetition in one segment [in your second screenshot] is definitely because of bug fixed in #1903.

Then I run 6 times every "r135", "r136 [aka #1838]" and "r135+#1903"
```
1. r135:             5 results out of 6 had very long repetition per one segment.
2. r136 [aka #1838]: 6 results out of 6 had very long repetition per one segment.
3. r135+#1903:       1 result  out of 6 had very long repetition per one segment.
```

That "1 result" from the "r135+#1903" tests was a bit different:

```
[00:17.780 --> 00:26.680]  GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGHGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG firmware
```

Note:
Results from "r136 [aka #1838]" looked better than from "r135" because I guess it was fixing other hallucinations.

**Reply by ioudove** [reply]
I am currently struggling with this https://github.com/openai/whisper/discussions/1962, but I have no programming knowledge. Even though Purfview has provided the correct solution and steps, I still haven't succeeded when attempting their method. I need to resolve this problem. Can someone please help me? If you are willing to assist, please simply tell me what I need to do. I sincerely appreciate any help you can provide.（Additionally, I have found that for the same video, if I use the medium model, this issue occurs much less frequently. However, when I use the large model, this problem often occurs.）
This is how I am running Whisper in Colab:
pip install -U openai-whisper
pip install setuptools-rust
!sudo apt update && sudo apt install ffmpeg
!whisper AAAAA.wav --model large --language zh --output_format srt

**Reply by Purfview** [reply]
> I am currently struggling with... Colab

Try:
```
!pip install git+https://github.com/Purfview/whisper/tree/patch-1.git
```

Or better use more advanced implementation, I don't know anyone who actually is using bare vanilla Whisper.

**Reply by ioudove** [reply]
Collecting git+https://github.com/Purfview/whisper/tree/patch-1.git
  Cloning https://github.com/Purfview/whisper/tree/patch-1.git to /tmp/pip-req-build-m0cyyocj
  Running command git clone --filter=blob:none --quiet https://github.com/Purfview/whisper/tree/patch-1.git /tmp/pip-req-build-m0cyyocj
  fatal: repository 'https://github.com/Purfview/whisper/tree/patch-1.git/' not found
  error: subprocess-exited-with-error
  
  × git clone --filter=blob:none --quiet https://github.com/Purfview/whisper/tree/patch-1.git /tmp/pip-req-build-m0cyyocj did not run successfully.
  │ exit code: 128
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× git clone --filter=blob:none --quiet https://github.com/Purfview/whisper/tree/patch-1.git /tmp/pip-req-build-m0cyyocj did not run successfully.
│ exit code: 128
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Reply by Purfview** [reply]
Try this:
`!pip install --force-reinstall git+https://github.com/openai/whisper.git@refs/pull/1903/head`

**Reply by ioudove** [reply]
First of all, thank you very much for your response and assistance. I appreciate it.

After running it, the returned result was:
 WARNING: Did not find branch or tag 'refs/pull/1903/head', assuming revision or ref.
 I am running Whisper in Colab. First，, I ran:
 pip install -U openai-whisper
 Then I ran:
 !pip install --force-reinstall git+https://github.com/openai/whisper.git@refs/pull/1903/head
 Did I do something wrong?

**Reply by Purfview** [reply]
> Did I do something wrong?

Probably command is not correct, try this:

`!pip install --force-reinstall --no-deps git+https://github.com/openai/whisper.git@1903`

**Reply by ioudove** [reply]
The returned result：WARNING: Did not find branch or tag '1903', assuming revision or ref.
 
 Has your patch not been integrated into the Whisper program yet?

**Reply by Purfview** [reply]
Mkay, try this:

```
!pip install --force-reinstall --no-deps git+https://github.com/openai/whisper.git@e7d46c9ed1df01f551d3b4bfb5f7e5ba310cfffb,2a7e12fd7addf40951ba372811d6db5c724b52b0
```

Or just the one commit [second one is not important]:
```
!pip install --force-reinstall --no-deps git+https://github.com/openai/whisper.git@e7d46c9ed1df01f551d3b4bfb5f7e5ba310cfffb
```

>Has your patch not been integrated into the Whisper program yet?

Yes, it's unmerged at the moment: https://github.com/openai/whisper/pull/1903

**Reply by ioudove** [reply]
I tested the second code you provided and it worked!
The result of the backtracking is
Thank you again ：D

Collecting git+https://github.com/openai/whisper.git@e7d46c9ed1df01f551d3b4bfb5f7e5ba310cfffb
  Cloning https://github.com/openai/whisper.git (to revision e7d46c9ed1df01f551d3b4bfb5f7e5ba310cfffb) to /tmp/pip-req-build-lkmfv6sc
  Running command git clone --filter=blob:none --quiet https://github.com/openai/whisper.git /tmp/pip-req-build-lkmfv6sc
  Running command git rev-parse -q --verify 'sha^e7d46c9ed1df01f551d3b4bfb5f7e5ba310cfffb'
  Running command git fetch -q https://github.com/openai/whisper.git e7d46c9ed1df01f551d3b4bfb5f7e5ba310cfffb
  Running command git checkout -q e7d46c9ed1df01f551d3b4bfb5f7e5ba310cfffb
  Resolved https://github.com/openai/whisper.git to commit e7d46c9ed1df01f551d3b4bfb5f7e5ba310cfffb
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: openai-whisper
  Building wheel for openai-whisper (pyproject.toml) ... done
  Created wheel for openai-whisper: filename=openai_whisper-20231117-py3-none-any.whl size=802828 sha256=32dea8a14a083c0952a45ad9304239e703c4e0112695060da2ce35cf5ed1c115
  Stored in directory: /root/.cache/pip/wheels/74/14/4b/e1a13def42450a9a91aadf52e5c1391c19cf30afde7f0d5953
Successfully built openai-whisper
Installing collected packages: openai-whisper
  Attempting uninstall: openai-whisper
    Found existing installation: openai-whisper 20231117
    Uninstalling openai-whisper-20231117:
      Successfully uninstalled openai-whisper-20231117
Successfully installed openai-whisper-20231117

**Reply by Purfview** [reply]
Did the patch helped with your issue?

**Reply by ioudove** [reply]
I'll test it in a couple days and I'll get back to you. Thanks again.

**Reply by ioudove** [reply]
I just tested it and still hallucinate. [My file](https://drive.google.com/file/d/1VrzAGrGUKhQhpyI5FDJd3_a9UoxQMZRx/view?usp=share_link) 
I'm running this in Colab.
**pip install -U openai-whisper
!pip install --force-reinstall --no-deps git+https://github.com/openai/whisper.git@e7d46c9ed1df01f551d3b4bfb5f7e5ba310cfffb
pip install setuptools-rust
!sudo apt update && sudo apt install ffmpeg
!whisper /content/drive/MyDrive/1111/210825青龙班-月线.wav --model large --language zh --output_format srt**

**Reply by Purfview** [reply]
Share srt with patch and srt without patch. And timings with hallucination.

**Reply by ioudove** [reply]
Since I don't have much free time at the moment, I won't be doing any more tests. Thanks again for your multiple replies and for helping me. Thank you.

**Comment by fillwithjoy1** [comment]
It's been a mixed bag. The ones that had problems were fixed, but the ones that were fine started hallucinating. It seems to be that static in something like Mandela Catalogue doesn't signal to the model to stop transcribing, and so it keeps repeating the last output. At least that's what I'm getting

Sent with Shortwave <https://www.shortwave.com?utm_medium=email&utm_content=signature&utm_source=am95c2luZ2guanMxQGdtYWlsLmNvbQ==>
